### DISCUSSION TITLE
passign custom model to tapActivity throws: instance of Spatie\Activitylog\Models\Activity given

### DISCUSSION THREAD

**Discussion by diptigajjar** [spatie/laravel-activitylog#1037] [Q&A]
Hello,

I've implemented the laravel-activitylog in my project and have extended the Activity Model in my custom Model UserActivityLog. But when I pass the UserActivityLog model to tapActivity function then it throws below error.

**Argument 1 passed to App\User::tapActivity() must be an instance of App\UserActivityLog, instance of Spatie\Activitylog\Models\Activity given**

**UserActivityLog model:**
class UserActivityLog extends Activity implements ActivityContract{
// code to list the activity 
}

**tapActivity:**
public function tapActivity(UserActivityLog $activity, string $eventName)
{
}

What causes this error?

Thank you.

**Comment by Gummibeer** [comment]
I'm pretty sure that you haven't configured your custom model in the package config - that's why the packages creates an instance of the default model.

**Reply by diptigajjar** [reply]
Hi @Gummibeer.
My config/activitylog.php file is like,

// 'activity_model' => \Spatie\Activitylog\Models\Activity::class,
    'activity_model' => \App\UserActivityLog::class,

I have copied the package config in the laravel config folder. Is this not the correct way?

**Reply by diptigajjar** [reply]
I used the tapActivity function to modify the changes. How can I prevent the log from this function conditionally?
